test/unit/migrations/037-test.js

```diff
@@ -0,0 +1,126 @@
+const assert = require('assert')
+let id = 0
+const migration37 = require('../../../app/scripts/migrations/037')
+const nullStorage = {meta: { version: 0 }, data: {}}
+const storage = {
+  meta: {},
+  data: { TransactionController: {
+    transactions: [],
+  } },
+}
+
+const counts = {}
+
+let status = [
+  'unapproved',
+  'rejected',
+  'approved',
+  'signed',
+  'submitted',
+  'confirmed',
+  'failed',
+  'dropped',
+]
+const nets = [ 1, 2, 3, 4, 5]
+let currentStatus = status.pop()
+let netIdIndex = 0
+
+const transactions160 = []
+const transactions60 = []
+
+do {
+  ++id
+  if (counts[currentStatus] === 20) currentStatus = status.pop()
+  transactions160.push(generateTxMeta(id, nets[netIdIndex], currentStatus))
+
+  if (!counts[currentStatus]) counts[currentStatus] = 1
+  else counts[currentStatus] += 1
+
+  if (id % 5 === 0) {
+    if (netIdIndex === 4) {
+      netIdIndex = 0
+    } else {
+      ++netIdIndex
+    }
+  }
+
+} while (id < 160)
+
+id = 0
+netIdIndex = 0
+const counts60 = {}
+status = [
+  'unapproved',
+  'approved',
+  'failed',
+  'submitted',
+
+  'rejected',
+  'signed',
+  'confirmed',
+  'dropped',
+]
+
+do {
+  ++id
+  // i only want unapproved
+  if (id % 10 === 0) currentStatus = status.pop()
+  if (counts60[currentStatus] === 5 && currentStatus === 'unapproved') currentStatus = status.pop()
+  if (counts60[currentStatus] === 5 && currentStatus === 'approved') currentStatus = status.pop()
+  if (counts60[currentStatus] === 4 && currentStatus === 'submitted') currentStatus = status.pop()
+  if (counts60[currentStatus] === 6 && currentStatus === 'failed') currentStatus = status.pop()
+
+  transactions60.push(generateTxMeta(id, nets[netIdIndex], currentStatus))
+  if (!counts60[currentStatus]) counts60[currentStatus] = 1
+  else counts60[currentStatus] += 1
+
+  if (id % 5 === 0) {
+    if (netIdIndex === 4) {
+      netIdIndex = 0
+    } else {
+      ++netIdIndex
+    }
+  }
+
+} while (id < 60)
+
+describe('storage is migrated successfully', () => {
+  it('should not fail', (done) => {
+    migration37.migrate(nullStorage).then((migratedData) => {
+      assert.equal(migratedData.meta.version, 37)
+      done()
+    }).catch(done)
+  })
+
+
+  it('should remove down to 40 transactions if their are over 40 transactions and their are enough final transactions to remove', (done) => {
```

Comment:
```suggestion
  it('should remove down to 40 transactions if there are over 40 transactions and there are enough final transactions to remove', (done) => {
```

Reply:
What is this test for? Why are there more than 40 transactions?

Reply:
I am somewhat confused by this question. Why wouldn't there be over 40 transactions? This migration's entire purpose is to trim the migration history down to 40.